src/lib.rs

```diff
@@ -72,13 +73,13 @@ impl BinaryInfo {
 
 /// Patches a binary to disable SIP.
 /// Right now it extracts x64 binary from fat/MachO binary and patches it.
-pub fn patch_binary(path: &str, output: &str) -> Result<()> {
-    let data = std::fs::read(path)?;
+pub fn patch_binary<P: AsRef<Path>>(path: P, output: P) -> Result<()> {
```

Comment:
Nitpick would split the path types for both inputs
```suggestion
pub fn patch_binary<P: AsRef<Path>, K: AsRef<Path>>(path: P, output: K) -> Result<()> {
```